pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -309,12 +309,12 @@ func (c *component) createOrPatch(ctx context.Context, sts *appsv1.StatefulSet,
 		Replicas:    &replicas,
 		ServiceName: c.values.PeerServiceName,
 		Selector: &metav1.LabelSelector{
-			MatchLabels: getCommonLabels(&c.values),
+			MatchLabels: getSelectors(&c.values),
```

Comment:
I think this change would require a recreation of the entire `StatefulSet`

Reply:
Yes, in the statefulset.go you should explicitly handle this case. If the selector has changed which will result in a new member not being able to be identified by the peer service then an STS recreation is required.

Reply:
Yes, recreation of statefulset is required. But without this future gardener upgrade (which handles label changes) won't be possible.

Reply:
Not necessarily, if there are default set of labels which are added to STS and the selector also has the same default set of labels then addition of labels only goes to STS and does not affect the discoverability of pods from the service.

Reply:
In the current change in my PR, the selector is only `instance: ETCDName`. So, no change in ETCD selector field will recreate sts. However, currently the selector fields contains "networking" labels. So, when this PR is merged and the new druid release goes in, recreation of STS is required to reflect the changes in existing STS

Reply:
Here again, why do we need to change this now? There has never been a problem reported by using the selector in the STS hat are used today. The PR description also says that it fixes the service selector, nothing else 🙁
Please revert this change.